frontend/pages/hosts/details/DeviceUserPage/ManualEnrollMdmModal/ManualEnrollMdmModal.tsx

```diff
@@ -1,51 +1,96 @@
-import React, { useState } from "react";
+import React, { useContext, useState } from "react";
+import { useQuery } from "react-query";
+import FileSaver from "file-saver";
 
+import { NotificationContext } from "context/notification";
+
+import DataError from "components/DataError";
 import Button from "components/buttons/Button";
 import Modal from "components/Modal";
+import Spinner from "components/Spinner";
 
 import mdmAPI from "services/entities/mdm";
 
 export interface IInfoModalProps {
   onCancel: () => void;
+  token: string;
 }
 
 const baseClass = "manual-enroll-mdm-modal";
 
-const ManualEnrollMdmModal = ({ onCancel }: IInfoModalProps): JSX.Element => {
-  const [showDownloading, setshowDownloading] = useState(false);
+const ManualEnrollMdmModal = ({
+  onCancel,
+  token,
+}: IInfoModalProps): JSX.Element => {
+  const { renderFlash } = useContext(NotificationContext);
+
+  const {
+    data: enrollmentProfile,
+    error: fetchMdmProfileError,
+    isFetching: isFetchingMdmProfile,
+  } = useQuery<string, Error>(
+    ["enrollment profile"],
+    () => mdmAPI.downloadDeviceUserEnrollmentProfile(token),
+    {
+      refetchOnWindowFocus: false,
+    }
+  );
 
-  const handleDownload = async () => {
-    setshowDownloading(true);
-    setTimeout(() => {
-      setshowDownloading(false);
-    }, 1000);
+  const onDownloadProfile = (evt: React.MouseEvent) => {
```

Comment:
Not sure if the requirements changed but I remember on the download button click there needs to be a 1000ms spinner to simulate loading and give the user a visual cue that something is happening.. is this still required?